modules/openapi-generator/src/main/resources/go-experimental/configuration.mustache

```diff
@@ -117,30 +178,116 @@ func (c *Configuration) AddDefaultHeader(key string, value string) {
 	for name, variable := range server.Variables {
-		if value, ok := variables[name]; ok {
-			found := bool(len(variable.EnumValues) == 0)
-			for _, enumValue := range variable.EnumValues {
-				if value == enumValue {
-					found = true
-				}
+			if value, ok := variables[name]; ok {
+					found := bool(len(variable.EnumValues) == 0)
+					for _, enumValue := range variable.EnumValues {
+							if value == enumValue {
+									found = true
+							}
+					}
+					if !found {
+							return "", fmt.Errorf("The variable %s in the server URL has invalid value %v. Must be %v", name, value, variable.EnumValues)
+					}
+					url = strings.Replace(url, "{"+name+"}", value, -1)
+			} else {
+					url = strings.Replace(url, "{"+name+"}", variable.DefaultValue, -1)
 			}
-			if !found {
-				return "", fmt.Errorf("The variable %s in the server URL has invalid value %v. Must be %v", name, value, variable.EnumValues)
+	}
+	return url, nil
+}
+
+// ServerUrl returns URL based on server settings
+func (c *Configuration) ServerUrl(index int, variables map[string]string) (string, error) {
+	return c.Servers.Url(index, variables)
+}
+
+func getServerIndex(ctx context.Context) (int, error) {
+	si := ctx.Value(ContextServerIndex)
+	if si != nil {
+		if index, ok := si.(int); ok {
+			return index, nil
+		}
+		return 0, reportError("Invalid type %T should be int", si)
+	}
+	return 0, nil
+}
+
+func getServerOperationIndex(ctx context.Context, endpoint string) (int, error) {
+	osi := ctx.Value(ContextOperationServerIndices)
+	if osi != nil {
+		if operationIndices, ok := osi.(map[string]int); !ok {
+			return 0, reportError("Invalid type %T should be map[string]int", osi)
+		} else {
+			index, ok := operationIndices[endpoint]
+			if ok {
+				return index, nil
 			}
-			url = strings.Replace(url, "{"+name+"}", value, -1)
+		}
+	}
+	return getServerIndex(ctx)
+}
+
+func getServerVariables(ctx context.Context) (map[string]string, error) {
+	sv := ctx.Value(ContextServerVariables)
+	if sv != nil {
+		if variables, ok := sv.(map[string]string); ok {
+			return variables, nil
+		}
+		return nil, reportError("Invalid type %T should be map[string]string", sv)
+	}
+	return nil, nil
+}
+
+func getServerOperationVariables(ctx context.Context, endpoint string) (map[string]string, error) {
+	osv := ctx.Value(ContextOperationServerVariables)
+	if osv != nil {
+		if operationVariables, ok := osv.(map[string]map[string]string); !ok {
+			return nil, reportError("Invalid type %T should be map[string]map[string]string", osv)
 		} else {
-			url = strings.Replace(url, "{"+name+"}", variable.DefaultValue, -1)
+			variables, ok := operationVariables[endpoint]
+			if ok {
+				return variables, nil
+			}
 		}
 	}
+	return getServerVariables(ctx)
+}
+
+// ServerUrlWithContext returns a new server URL given an endpoint
+func (c *Configuration) ServerUrlWithContext(ctx context.Context, endpoint string) (string, error) {
+	if ctx == nil {
+		return c.BasePath, nil
```

Comment:
Alternatively we can replace it with `return sc.Url(0, nil)` and move it after line 275.

Reply:
I have changed it to `return sc.Url(0, nil)` and removed `BasePath` completely.